multiuser/keycloak/che-multiuser-keycloak-user-remover/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakPasswordGrantTokenRequester.java

```diff
@@ -0,0 +1,135 @@
+   * @param keycloakTokenEndpoint the Keycloak token endpoint
+   * @return Keycloak access token
+   * @throws ServerException when any server error occurs while performing the request
+   * @throws IOException when an IO error occurs while connecting to the server
+   * @throws JsonSyntaxException when an error occurs while parsing JSON response
+   */
+  public String requestToken(
+      String keycloakUser, String keycloakPassword, String keycloakTokenEndpoint)
+      throws ServerException, IOException, JsonSyntaxException {
+
+    String accessToken = "";
+    HttpURLConnection http = null;
+    try {
+      http = createURLConnection(keycloakUser, keycloakPassword, keycloakTokenEndpoint);
+      checkResponseCode(http, keycloakTokenEndpoint);
+
+      final BufferedReader response =
+          new BufferedReader(new InputStreamReader(http.getInputStream(), UTF_8));
+      accessToken = getAccessToken(response);
+
+    } catch (IOException | JsonSyntaxException ex) {
+      throw ex;
+    } finally {
+      if (http != null) {
+        http.disconnect();
+      }
+    }
+    return accessToken;
+  }
+
+  private HttpURLConnection createURLConnection(
+      String keycloakUser, String keycloakPassword, String keycloakTokenEndpoint)
+      throws IOException {
+    HttpURLConnection http = (HttpURLConnection) new URL(keycloakTokenEndpoint).openConnection();
+    http.setConnectTimeout(60000);
+    http.setReadTimeout(60000);
+    http.setRequestMethod(POST);
+    http.setAllowUserInteraction(false);
+    http.setRequestProperty(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED);
+    http.setInstanceFollowRedirects(true);
+    http.setDoOutput(true);
+    StringBuilder sb = new StringBuilder();
+    sb.append(GRANT_TYPE + "=" + GRANT_TYPE_VALUE)
+        .append("&" + USERNAME + "=")
+        .append(keycloakUser)
+        .append("&" + PASSWORD + "=")
+        .append(keycloakPassword)
+        .append("&" + CLIENT_ID + "=" + CLIENT_ID_VALUE);
+
+    try (OutputStream output = http.getOutputStream()) {
+      output.write(sb.toString().getBytes(UTF_8));
+    }
+    return http;
+  }
+
+  private void checkResponseCode(HttpURLConnection http, String keycloakTokenEndpoint)
+      throws IOException, ServerException {
+    if (http.getResponseCode() != 200) {
+      throw new ServerException(
+          "Cannot get Keycloak access token. Server response: "
+              + keycloakTokenEndpoint
+              + " "
+              + http.getResponseCode()
+              + IoUtil.readStream(http.getErrorStream()));
+    }
+  }
+
+  private String getAccessToken(BufferedReader response) throws ServerException {
+    JsonParser jsonParser = new JsonParser();
+    JsonElement jsonElement = jsonParser.parse(response);
+    JsonObject asJsonObject = jsonElement.getAsJsonObject();
+    if (!asJsonObject.has(ACCESS_TOKEN)) {
+      throw new ServerException("Keycloak access token does not exist.");
```

Comment:
I'm not sure if this is the correct exception to throw here

Reply:
technically this situation should never occur, and since we already throw ServerException in the `checkResponseCode` I think it is fine to throw it here also